src/main/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatchCasesV3.java

```diff
@@ -0,0 +1,273 @@
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+    // Approve dataset GRU condition
+    if (getNullableOrFalse(dataset.getGeneralUse())) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+    // Approve dataset HMB condition
+    if (getNullableOrFalse(dataset.getHmbResearch())) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+    // short-circuit if no diseases specified in the purpose, but the dataset specifies diseases
+    if (!dataset.getDiseaseRestrictions().isEmpty() && purpose.getDiseaseRestrictions().isEmpty()) {
+      return MatchResult.from(MatchResultType.DENY, Collections.singletonList(DS_F1));
+    }
+
+    // We want all-purpose disease IDs to be a subclass of any dataset disease ID
+    List<String> failures = new ArrayList<>();
+    for (Map.Entry<String, List<String>> entry : purposeDiseaseIdMap.entrySet()) {
+      boolean match = entry.getValue()
+          .stream()
+          .anyMatch(dataset.getDiseaseRestrictions()::contains);
+      if (!match) {
+        failures.add(String.format(DS_F2,  entry.getKey()));
+      }
+    }
+
+    return MatchResult.from(failures.isEmpty() ? MatchResultType.APPROVE : MatchResultType.DENY, failures);
+  }
+
+  /**
+   * RP: HMB
+   * Datasets:
+   *      Any dataset tagged with GRU
+   *      Any dataset tagged with HMB
+   *
+   * @param purpose The data use object representing the Research Purpose
+   * @param dataset The data use object representing the Dataset
+   */
+  static MatchResult matchHMB(DataUseV3 purpose, DataUseV3 dataset) {
+    // short-circuit hmb if not set
+    if (Objects.isNull(purpose.getHmbResearch())&& (Objects.isNull(dataset.getHmbResearch()))) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    boolean purposeGRU = getNullableOrFalse(purpose.getGeneralUse());
+    boolean purposeHMB = getNullableOrFalse(purpose.getHmbResearch());
+    boolean datasetGRU = getNullableOrFalse(dataset.getGeneralUse());
+    boolean datasetHMB = getNullableOrFalse(dataset.getHmbResearch());
+
+    // Deny purpose GRU condition
+    if (datasetHMB && purposeGRU) {
+      return MatchResult.from(MatchResultType.DENY, Collections.singletonList(HMB_F1));
+    }
+
+    // Approve dataset GRU condition
+    if (datasetGRU) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    }
+
+    // Approve dataset HMB condition
+    if (purposeHMB && datasetHMB) {
+      return MatchResult.from(MatchResultType.APPROVE, Collections.emptyList());
+    } else {
+      return MatchResult.from(MatchResultType.DENY, Collections.singletonList(HMB_F2));
+    }
+  }
+
+  /**
+   * RP: Study population origins or ancestry
+   * Future use is limited to research involving a specific population [POA]
+   * Datasets:
+   *      Any dataset tagged with GRU
+   *      Any dataset tagged with POA
```

Comment:
There is a deny case on the doc: 
> Deny if the dataset's Primary DUO terms are DS-, HMB

And the tests work in that case, we should document that here.